src/classes/TDTM_ObjectDataGateway.cls

```diff
@@ -91,49 +91,16 @@ public without sharing class TDTM_ObjectDataGateway implements TDTM_iTableDataGa
      */
     public static list<Trigger_Handler__c> listTH {
         get {
-            if (listTH == null) {
-                listTH =  [select Asynchronous__c, Class__c, Load_Order__c, Object__c, Active__c, Usernames_to_Exclude__c, 
-                    Trigger_Action__c from Trigger_Handler__c order by Load_Order__c];
+            if(Advancement_Info.useAdv() && Advancement_Info.getApiNPSP() != null) {
```

Comment:
I think it is safer to write it in this way.  Basically, Adv need to have the full control about what "listTH" gets return to NPSP if Adv is installed. Even the listTH is not empty, we still pass it into Adv class. And in Adv class will decide whether pass it back or re query it. 

1. Since listTH is a public static variable, it is possible that someone set it directly from the code. If we just listTH if listTH != null, then Adv will lose the ability to filter it. 

2. The code is more clear to other developer. We just say, if Adv is installed then let Adv handle it. Otherwise just the original NPSP code. So, we can be sure that NPSP will not break if Adv is not installed.